### ISSUE TITLE
Failing for java 8 update 252

### ISSUE THREAD

**Issue by ankurga** [dependency-check/DependencyCheck#6162]
**Describe the bug**
We have java version `Java version: 1.8.0_252` but it still fails saying `dependency-check requires Java 8 update 251 or higher`

**Version of dependency-check used**
The problem occurs using version 9.0.1 of the maven plugin.

**Log file**

```bash
go@myserver:/var/lib/go-agent-2/pipelines/demo-hello-world$ mvn --version
Apache Maven 3.6.0
Maven home: /usr/share/maven
Java version: 1.8.0_252, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "4.19.0-25-cloud-amd64", arch: "amd64", family: "unix"
go@integration-agarg:/var/lib/go-agent-2/pipelines/demo-hello-world$ mvn --batch-mode clean org.owasp:dependency-check-maven:9.0.1:check verify -Pcoverage -DossindexAnalyzerEnabled=false -DnvdApiKey=xxxx
[INFO] Scanning for projects...
[INFO]
[INFO] --------------------< com.oscato:demo-hello-world >---------------------
[INFO] Building demo-hello-world 0.7.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ demo-hello-world ---
[INFO]
[INFO] --- dependency-check-maven:9.0.1:check (default-cli) @ demo-hello-world ---
[ERROR] Non-supported Java Runtime: dependency-check requires at least Java 8 update 251 or higher.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.572 s
[INFO] Finished at: 2023-11-28T09:58:11Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.owasp:dependency-check-maven:9.0.1:check (default-cli) on project demo-hello-world: Execution default-cli of goal org.owasp:dependency-check-maven:9.0.1:check failed: dependency-check requires Java 8 update 251 or higher -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

**To Reproduce**

Run: `mvn --batch-mode clean org.owasp:dependency-check-maven:9.0.1:check verify -Pcoverage -DossindexAnalyzerEnabled=false -DnvdApiKey=xxxx` command in any maven project.

**Expected behavior**
Should pass or give a different error as we already have a higher version than minimum required for java 8.

**Comment by jeremylong** [comment]
Could you run `java --version` and provide the output?

**Comment by lbillinghamwrk** [comment]
I think that this may be because debian java builds add extra `-` delimited things to the version number

**Comment by jeremylong** [comment]
Sure - but I need an example so I can fix the parsing of the java version.

**Comment by jeremylong** [comment]
running `mvn --version` is not what I need to see. I need to see `java --version`.

**Comment by ankurga** [comment]
> Could you run `java --version` and provide the output?

```bash
$ java -version
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-8u252-b09-1~deb9u1-b09)
OpenJDK 64-Bit Server VM (build 25.252-b09, mixed mode)
```

We also tried to add a println to debug in this file: https://github.com/jeremylong/DependencyCheck/blob/ed3912cf149a95a6d54401456bd4dac3e847cf88/core/src/main/java/org/owasp/dependencycheck/utils/Utils.java#L57

And we got below result:

`RuntimeVERsion:1.8.0_252-8u252-b09-1~deb9u1-b09`

**Comment by jeremylong** [comment]
Perfect, thanks.

**Comment by lbillinghamwrk** [comment]
> closed this as complete in #6163 1 minute ago

super speedy ❤️ thank, you
